### DISCUSSION TITLE
the code didn't work

### DISCUSSION THREAD

**Discussion by yiyu001** [gdsfactory/gdsfactory#2091] [Q&A] [answered]
the code in the gdsfactory is still useful,but when i use the code in tutorial draw the polygon:
import gdsfactory as gf
from gdsfactory.generic_tech import get_generic_pdk

gf.config.rich_output()
gf.CONF.display_type = 'klayout'

PDK = get_generic_pdk()
PDK.activate()

@gf.cell
def demo_polygons():
    # Create a blank component (essentially an empty GDS cell with some special features)
    c = gf.Component()

    # Create and add a polygon from separate lists of x points and y points
    # (Can also be added like [(x1,y1), (x2,y2), (x3,y3), ... ]
    poly1 = c.add_polygon(
        [(-8, 6, 7, 9), (-6, 8, 17, 5)], layer=1
    )  # GDS layers are tuples of ints (but if we use only one number it assumes the other number is 0)
    return c

c = demo_polygons()
c.plot()  # show it in KLayout
the klayout didn't response.

**Comment by yiyu001** [comment]
the terminal shows that:
PS D:\Program Files\Python311\Lib\site-packages\gdsfactory> & "D:/Program Files/Python311/python.exe" "d:/Program Files/Python311/Lib/site-packages/gdsfactory/components/new_polygon_try"
2023-09-17 21:16:02.714 | INFO     | gdsfactory.technology.layer_views:__init__:790 - Importing LayerViews from YAML file: 'D:\\Program Files\\Python311\\Lib\\site-packages\\gdsfactory\\generic_tech\\layer_views.yaml'.
2023-09-17 21:16:02.714 | INFO     | gdsfactory.pdk:activate:334 - 'generic' PDK is now active
PS D:\Program Files\Python311\Lib\site-packages\gdsfactory> 
and nothing happen

**Reply by joamatab** [reply]
Are you running on a jupyter notebook? 

c.plot() is for showing image in notebooks 



c.show() for showing in klayout if you have klive plug-in installed 

Otherwise you need to run

c.write_gds(path) and then open file in klayout

**Comment by joamatab** [comment, marked as answer]
Are you running on a jupyter notebook?

c.plot() is for showing image in notebooks

c.show() for showing in klayout if you have klive plug-in installed

Otherwise you need to run

c.write_gds(path) and then open file in klayout

**Comment by yiyu001** [comment]
I used vscode to run the code，do i need to download the jupyter notebook to run the code？



---Original---
From: "Joaquin ***@***.***&gt;
Date: Mon, Sep 18, 2023 08:42 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [gdsfactory/gdsfactory] the code didn't work (Discussion #2091)




 
Are you running on a jupyter notebook?
 
c.plot() is for showing image in notebooks
 
c.show() for showing in klayout if you have klive plug-in installed
 
Otherwise you need to run
 
c.write_gds(path) and then open file in klayout

**Comment by yiyu001** [comment]
I have solved the problem，thank you very much！



---Original---
From: "Joaquin ***@***.***&gt;
Date: Mon, Sep 18, 2023 08:42 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [gdsfactory/gdsfactory] the code didn't work (Discussion #2091)




 
Are you running on a jupyter notebook?
 
c.plot() is for showing image in notebooks
 
c.show() for showing in klayout if you have klive plug-in installed
 
Otherwise you need to run
 
c.write_gds(path) and then open file in klayout
